bears/general/IndentationBear.py

```diff
@@ -386,6 +399,23 @@ def get_valid_sequences(file, sequence, annotation_dict):
                    lt_eq(sequence_position, comment.end)):
                     valid = False
 
+                if check_ending:
+                    if(comment.start.line == sequence_position.line and
+                            comment.end.line == sequence_position.line):
+                        sequence_line_text = sequence_line_text[
+                            :comment.start.column - 1] + sequence_line_text[
+                            comment.end.column-1:]
+
+            if encapsulators:
+                for encapsulator in encapsulators:
+                    if(gt_eq(sequence_position, encapsulator.start) and
```

Comment:
Lots of levels of nesting in this area :(

If you are using similar constructs to this elsewhere, it might be worth it to make a function `overlaps()` or `inside_range()`. in that case you could use `filter()` here instead of the loop and simplify everything.

These are just dieas though. Also, if encapsulators would be `[]` or `()` instead of `None` you wouldn't need the if statement at the top.